### DISCUSSION TITLE
Spreadsheet for mapping datasets to the TERN ontology

### DISCUSSION THREAD

**Discussion by gwittich** [ternaustralia/ontology_tern#73] [General] [answered]
Javier talked us through an amazing Spreadsheet for mapping datasets to the TERN ontology during the Hackathon (ecoplots_vocabulary_mappings). Has that spreadsheet been updated to reflect the changes to the ontology as published early December 2021?
If so, is it possible for Gaia Resources to have a copy?

**Comment by smguru** [comment, marked as answer]
Hello Gail, we can provide you the spreadsheet but that spreadsheet was
built to map TERN surveillance data, @javiersan18  can you please send the mapping spreadsheet to Gail.

regards
Guru

**Reply by javiersan18** [reply]
Hello Gail,

As Guru said, this mapping is for 1 dataset (TERN surveillance) and it is very specific to our implementation.
We have written a python ETL which reads both the data (DB) and the mapping (spreadsheet) and generates RDF output for:
- Samples / Sampling / FeatureOfInterest
- Observation / Attribute / Instrument
- Other classes in the Tern Ontology related to them... e.g. Results

Creation of Site/SiteVisit is created by the ETL but it has nothing to do with the mapping.
Basically, the mapping is telling the ETL... this column in this table is an observation, belongs to that FOI, and has all these properties.

Here is the link:
https://docs.google.com/spreadsheets/d/1YPiYmec9x4ZfeomYqH5Ex-2F1wIrQKtoofYy8r19rkY/edit?usp=sharing

I am happy to resolve any doubt about the mapping.

Best regards,
Javier Sanchez

**Reply by gwittich** [reply]
Thanks Javier and Guru. I will have a look and see how I go...
